src/module.c

```diff
@@ -1003,16 +1012,39 @@ JL_DLLEXPORT jl_value_t *jl_module_names(jl_module_t *m, int all, int imported)
         jl_sym_t *asname = b->globalref->name;
         int hidden = jl_symbol_name(asname)[0]=='#';
         int main_public = (m == jl_main_module && !(asname == jl_eval_sym || asname == jl_include_sym));
-        if ((b->publicp ||
+        if ((jl_atomic_load_relaxed(&b->value) != (jl_value_t*)m) &&
```

Comment:
I think this should be based on the name instead of the value?

Reply:
It seems that this code was added in #42092, but I think it can be removed without any issues, so I plan to do that.